src/search/landmarks/landmark_count_heuristic.cc

```diff
@@ -37,8 +37,11 @@ LandmarkCountHeuristic::LandmarkCountHeuristic(const options::Options &opts)
           admissible ||
           (!task_properties::has_axioms(task_proxy) &&
            (!task_properties::has_conditional_effects(task_proxy) || conditional_effects_supported))),
+      log(utils::get_log_from_options(opts)),
```

Comment:
I saw in every heuristic that the log option is added to its parsing function and to its constructor. Should we move this just to the base class `Heuristic` or even `Evaluator`?

Reply:
Yes, I think having the log in the `Heuristic` class should be the goal. But I would only do it once all heuristics actually use a local log instead of `g_log`. I don't think `Evaluator` should have a log. None of the existing evaluators produce output themselves. Only the base class has two `print` methods which are called by the search with the log passed as a parameter by the search. I think this makes sense because search can decide on its own if to call this method and how to configure the log for that.

Reply:
The distinction "heuristic" vs. "evaluator" is not really that meaningful any more. The names don't match what the difference between them is. The class "Heuristic" now is the class "Evaluator" with some extra bits and pieces that don't really belong together and that are not that closely related to an intuitive distinction between what a heuristic is vs. what a generic evaluator is. We did a lot of refactoring there, but didn't yet complete it. In the meantime, I would avoid adding more distinctions between "Evaluator" and "Heuristic". If the solution for now is to put all logging into the individual heuristics, that works for me.

Reply:
In that case I would prefer having the log in the Evaluator base class. Adding one to each heuristic would be very tedious.

Reply:
Works for me.